pkg/stats/io_snapshots.go

```diff
@@ -37,25 +42,78 @@ func (h *snapshotHistory) SetInput(id string, latestInput json.Marshaler) {
 	h.latestInput[id] = latestInput
 }
 
-func (h *snapshotHistory) GetInput() ([]byte, error) {
+func (h *snapshotHistory) GetInput(format string, clusters []string, namespaces []string, resourceTypes []string) ([]byte, error) {
 	h.lock.RLock()
 	defer h.lock.RUnlock()
 	if h.latestInput == nil {
 		return []byte{}, nil
 	}
-	result := map[string]interface{}{}
-	for id, obj := range h.latestInput {
-		jsn, err := obj.MarshalJSON()
-		if err != nil {
-			return nil, err
+	genericMaps, err := getGenericMaps(h.latestInput)
```

Comment:
Discussed this over phone with John but in general I went with this approach because we are already taking a full copy of the snapshot initially so we don't have to lock it long. Working with that copy to create the result I believe is more memory efficient than working with another new object for the result.